### PR TITLE
Convert networkservicemesh into a Ligato plugin

### PR DESCRIPTION
**Pull request by mestery**
This series of commits converts networkservicemesh into a Ligato plugin. This allows us to take advantage of a few things which Ligato provides:

- Logging
- Common front-end APIs
- gRPC and REST APIs

See the RUN.md for instructions on how to get this up and running. The code comes up as a Ligato plugin and is able to idle. Further testing for the device-plugin backend integration with Kubernetes is required.

Would like feedback from @fkautz @edwarnicke @jhnbrns on this approach, etc.

### CODE REVIEW

File: cmd/nsm/main.go [hunk 1]
```diff
@@ -1,38 +1,27 @@
...
 
+// netmesh main entry point.
 func main() {
-	var wg sync.WaitGroup
-
-	log.SetFlags(log.Flags() | log.Lshortfile)
-	log.Println("Starting NSM")
-
-	dp := nsmdp.NewNSMDevicePlugin()
-	dp.Serve()
```
Comment (mestery) [inline cmd/nsm/main.go]: This logic moves into the AfterInit() function in plugins/netmesh/plugin_impl_netmesh.go.

File: cmd/nsm/main.go [hunk 2]
```diff
@@ -1,38 +1,27 @@
...
-		syscall.SIGINT,
-		syscall.SIGTERM,
-		syscall.SIGQUIT)
-	wg.Add(1)
-
-	go func() {
-		defer wg.Done()
-		s := <-sigChan
-		log.Printf("Received signal \"%v\", shutting down.", s)
-		dp.Stop()
```
Comment (mestery) [inline cmd/nsm/main.go]: This moves into the Close() function in plugins/netmesh/plugin_impl_netmesh.go.

**Comment by mestery** [comment]
@edwarnicke There is a lot going on here. :) What I'd like to propose is that I look at breaking this down a bit so that we utilize the Ligato plugin infra for control loop type code. Once that is done, I'll post a new PR. That's the immediate area where I believe we can gain some benefit.

**Comment by edwarnicke** [comment]
Sounds good to me, @fkautz thoughts?

**Comment by mestery** [comment]
Closing in lieu of PR46: https://github.com/ligato/networkservicemesh/pull/46

**Pull request closed without merging**